### PR TITLE
Des 3542

### PR DESCRIPTION
**Pull request by matthewgardner**
Add popup to support Basic Authentication in Paragon/Minuet
https://perzoinc.atlassian.net/browse/DES-3542

### CODE REVIEW

File: minuet/symphony/Paragon.Runtime/CefHandlers/CefWebRequestHandler.cs [hunk 1]
```diff
@@ -1,9 +1,35 @@
...
+using Paragon.Runtime.WinForms;
+using System.Windows.Forms;
+using System.Diagnostics;
+using System.Windows;
+using System.Windows.Interop;
+
 
 namespace Paragon.Runtime
 {
+    public class WindowWrapper : System.Windows.Forms.IWin32Window
```
Comment (lneir) [inline minuet/symphony/Paragon.Runtime/CefHandlers/CefWebRequestHandler.cs:13]: Is there a reason why we are using win forms versus WPF?  Outside of screen capture the rest of the app is wpf based, you'll get better styling support with wpf.  In future we should avoid using win forms everywhere that way we don't have to take dependency on additional DLLs and have consistency in UI.

File: minuet/symphony/Paragon.Runtime/CefHandlers/CefWebRequestHandler.cs [hunk 2]
```diff
@@ -47,5 +73,27 @@ protected override bool OnCertificateError(CefBrowser browser, CefErrorCode cert
             Logger.Error("Failed to load resource due to an invalid certificate: " + requestUrl);
             return base.OnCertificateError(browser, certError, requestUrl, sslInfo, callback);
         }
+        protected override bool GetAuthCredentials(CefBrowser browser, CefFrame frame, bool isProxy, string host, int port, string realm, string scheme, CefAuthCallback callback)
+        {
+            string strFriendlyName = AppDomain.CurrentDomain.FriendlyName;
+            Process[] pro = Process.GetProcessesByName(strFriendlyName.Substring(0, strFriendlyName.LastIndexOf('.')));
+            System.Windows.Forms.IWin32Window handle = new WindowWrapper(pro[0].MainWindowHandle);
```
Comment (lneir) [inline minuet/symphony/Paragon.Runtime/CefHandlers/CefWebRequestHandler.cs:80]: Are we sure this is the best way to get window handle?  Looks a bit fragile.   You can also get window handle for main browser from: CefBrowser browser.

File: minuet/symphony/Paragon.Runtime/CefHandlers/CefWebRequestHandler.cs (hunk 1 above)
Comment (matthewgardner) [inline minuet/symphony/Paragon.Runtime/CefHandlers/CefWebRequestHandler.cs:13]: Yes - I am a java developer with no clue - we will rewrite this :)

File: minuet/symphony/Paragon.Runtime/CefHandlers/CefWebRequestHandler.cs (hunk 2 above)
Comment (matthewgardner) [inline minuet/symphony/Paragon.Runtime/CefHandlers/CefWebRequestHandler.cs:80]: We can revisit with rewrite to WPF

**Pull request closed without merging**